File: OnlineStore.Bll/Services/DefaultServices/OrderService.cs
using OnlineStore.Bll.Services.Interfaces;
using OnlineStore.Common.Models;
using OnlineStore.Dal.Repositories.Interfaces;
using System.Collections.Generic;
using System.Threading.Tasks;

namespace OnlineStore.Bll.Services.DefaultServices
{
    public class OrderService : IOrderService
    {
        private readonly IOrderRepository _orderRepository;

        public OrderService(IOrderRepository orderRepository)
        {
            _orderRepository = orderRepository;
        }

        public async Task<Order> Create(Order order)
        {
            return await _orderRepository.Create(order);
        }

        public async Task Delete(int id)
        {
            await _orderRepository.Delete(id);
        }

        public async Task<IEnumerable<Order>> GetAll()
        {
            return await _orderRepository.GetAll();
        }

        public async Task<Order> GetById(int id)
        {
            return await _orderRepository.GetById(id);
        }

        public async Task<IEnumerable<Order>> GetByUserId(int userId)
        {
            return await _orderRepository.GetByUserId(userId);
        }

        public async Task<Order> Update(Order order)
        {
            return await _orderRepository.Update(order);
        }
    }
}

File: OnlineStore.Web/ClientApp/src/app/services/order.service.ts
import { HttpClient, HttpHeaders } from "@angular/common/http";
import { Injectable } from "@angular/core";
import { Observable } from "rxjs";
import { Order } from "../models/order";
import { OrderData } from "../models/order-data";

@Injectable()
export class OrderService {
  private readonly path: string = 'api/order';

  constructor(private readonly http: HttpClient) { }

  public create(order: Order): Observable<Order> {
    return this.http.post<Order>(this.path, order);
  }

  public delete(id: number): Observable<void> {
    const options = {
      headers: new HttpHeaders({
        'Content-Type': 'application/json',
      }),
      body: {
        id: id
      },
    };

    return this.http.delete<void>(this.path, options);
  }

  public getAll(): Observable<OrderData[]> {
    return this.http.get<OrderData[]>(this.path);
  }

  public getById(id: number): Observable<Order> {
    let path: string = this.path + "/" + id;

    return this.http.get<Order>(path);
  }

  public update(order: Order): Observable<Order> {
    return this.http.put<Order>(this.path, order);
  }

  public getByUserId(userId: number): Observable<OrderData[]> {
    let path = this.path + "/byUser/" + userId;

    return this.http.get<OrderData[]>(path);
  }

  public getNotOrdered(): Observable<Order> {
    let path = this.path + 'notOrdered';

    return this.http.get<Order>(path);
  }
}
